From a97c605eb9fb9a871fbc06098507c318c4f1c7d6 Mon Sep 17 00:00:00 2001
From: AKASHI Takahiro <takahiro.akashi@linaro.org>
Date: Fri, 28 Feb 2020 09:06:34 +0900
Subject: [PATCH 1/1] common: update_tftp: remove unnecessary build check

Logically, the current update_tftp() should and does compile and work
correctly even without satisfying the following condition:

> #if defined(CONFIG_UPDATE_TFTP) && !defined(CONFIG_MTD_NOR_FLASH)
> #error "CONFIG_UPDATE_TFTP and !CONFIG_MTD_NOR_FLASH needed for
>  legacy behaviour"
> #endif

It would be better to just drop it so that this function will be
used on wider range of platforms.

Signed-off-by: AKASHI Takahiro <takahiro.akashi@linaro.org>
---
 common/update.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/common/update.c b/common/update.c
index c8dd346a09..ade029851d 100644
--- a/common/update.c
+++ b/common/update.c
@@ -14,10 +14,6 @@
 #error "CONFIG_FIT and CONFIG_OF_LIBFDT are required for auto-update feature"
 #endif
 
-#if defined(CONFIG_UPDATE_TFTP) && !defined(CONFIG_MTD_NOR_FLASH)
-#error "CONFIG_UPDATE_TFTP and !CONFIG_MTD_NOR_FLASH needed for legacy behaviour"
-#endif
-
 #include <command.h>
 #include <env.h>
 #include <flash.h>
@@ -210,8 +206,9 @@ static int update_flash(ulong addr_source, ulong addr_first, ulong size)
 		printf("Error: could not protect flash sectors\n");
 		return 1;
 	}
+#else
+	return -1;
 #endif
-	return 0;
 }
 
 static int update_fit_getparams(const void *fit, int noffset, ulong *addr,
-- 
2.26.2

